src/plugins/layout/shadowdom.ts

```diff
@@ -240,29 +214,18 @@ export class ShadowDom {
     };
 
     // Collect all new nodes in the top-down order
-    let topNewNodes = this.doc.getElementsByClassName(NewNodeClassName);
-    while (topNewNodes.length > 0) {
-      let topNode = topNewNodes[0] as IShadowDomNode;
-      let discoverQueue: IShadowDomNode[] = [ topNode ];
-      while (discoverQueue.length > 0) {
-        let next = discoverQueue.shift();
-        if (this.hasClass(next, NewNodeClassName)) {
-          summary.newNodes.push(next);
-          this.removeClass(next, NewNodeClassName);
-          // Add children right-to-left
-          let children = next.childNodes;
-          for (let i = children.length - 1; i >= 0; i--) {
-            discoverQueue.push(children[i] as IShadowDomNode);
-          }
-        }
-      }
+    let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
```

Comment:
>getElementsByClassName [](start = 28, length = 22)

since we're in here, maybe add TODOs on the getElementsByCLassName indicating we need to get off of Live HTMLCollections